filters/openpolicyagent/openpolicyagent.go

```diff
@@ -470,6 +485,42 @@ func (opa *OpenPolicyAgentInstance) MetricsKey(key string) string {
 	return key + "." + opa.bundleName
 }
 
+type bufferedReadCloser struct {
+	reader *bufio.Reader
+	closer io.ReadCloser
+}
+
+func (r *bufferedReadCloser) Read(p []byte) (n int, err error) {
+	return r.reader.Read(p)
+}
+
+func (r *bufferedReadCloser) Close() error {
+	return r.closer.Close()
+}
+
+func (opa *OpenPolicyAgentInstance) ExtractHttpBodyOptionally(req *http.Request) (io.ReadCloser, []byte, error) {
+	body := req.Body
+	var rawBody []byte
+	if body != nil && !opa.EnvoyPluginConfig().SkipRequestBodyParse && opa.maxBodyBytes > 0 {
+		bufferedReader := bufio.NewReaderSize(body, opa.maxBodyBytes)
```

Comment:
This will always allocate maxBodyBytes while actual body could be much smaller

Reply:
Correct, it saved re-implementing a smarter buffered reader which I can also do. My assumption was that the buffer is typically in the range of hundreds of KBs so should not hurt too much. Would you recommend re-doing this with a fixed size buffer (that is in low two digits KBs)?